### PR TITLE
Add comprehensive unit test suite (132 tests)

### PR DESCRIPTION
**Pull request by voteblake**
## Summary
Adds a complete unit test suite with 132 tests covering all major modules:

- **test_beam_search.py** (51 tests): Beam search algorithm invariants (pruning, expansion, context propagation, termination)
- **test_branch.py** (3 tests): Branch dataclass serialization
- **test_data_loader.py** (21 tests): Data loading and splitting utilities
- **test_evaluate.py** (15 tests): ROUGE evaluation and report aggregation
- **test_evaluator.py** (16 tests): TreeOfThoughtEvaluator scoring modes
- **test_graph_env_tools.py** (13 tests): GraphEnvironment tool methods
- **test_rate_limit.py** (13 tests): Rate limit retry behavior
- **conftest.py**: Shared fixtures and factories

## Key Features
- **Zero LLM/API calls**: All tests use mocks and fixtures; no external dependencies
- **No filesystem I/O**: Uses tmp_path and in-memory data structures
- **Algorithm verification**: Tests enforce beam search invariants from arXiv:2502.13247
- **Edge case coverage**: Empty inputs, boundary conditions, type validation
- **Pytest configuration**: Added `[tool.pytest.ini_options]` to pyproject.toml

## Bug Fix
Fixed potential IndexError in GraphToTSolver when current_beam is empty by adding bounds check before accessing `current_beam[0]`.

### CODE REVIEW

**Comment by voteblake** [comment]
@codex, please review the PR - are these all good and necessary tests of our library's behavior and the paper's invariants? Is there anywhere where we are essentially just testing mocks or testing underlying libraries rather than our own logic?

File: tests/test_beam_search.py [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+# ---------------------------------------------------------------------------
+# Helpers
+# ---------------------------------------------------------------------------
+
+
+def _scored_branches(*score_answer_pairs):
+    """Build a list of branches pre-sorted by score descending."""
+    branches = [make_branch(answer=ans, score=sc) for sc, ans in score_answer_pairs]
+    return sorted(branches, key=lambda b: b.score, reverse=True)
```
[the remark this answers is not in the document]
Comment (voteblake) [reply tests/test_beam_search.py]: @codex Is this just a matter of removing sorted() here or do I need to intentionally un-sort every call?
